From 9771fa00ba14a9514a6d26832a35b8b92547d6d0 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Fri, 31 Jan 2025 16:22:04 +0100
Subject: [PATCH 1497/1695] dts: stm32: add Unique device ID (UID) for
 stm32mp21

Unique device IDentifier (UID) is 96 bits and is defined in OTP5-7, as:
OTP5 UID[31:0] Unique ID bits,
OTP6 UID[63:32] Unique ID bits,
OTP7 UID[95:64] Unique ID bits.

It is used by ADAC (Authentication Debug Access Control) to respond to
ADAC_DISCOVERY_CMD, used by the debugger to request some information about
the target.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git fbbebbefccbbeffeeb3d5512e3a5f908643568da dts: stm32: add Unique device ID (UID) for stm32mp21]
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Change-Id: Id68e2b65020fc98545b05b8475d266aad9cadaaa
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/433595
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/dts/stm32mp211.dtsi | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp211.dtsi b/core/arch/arm/dts/stm32mp211.dtsi
index bf354fc06..dc8c246d1 100644
--- a/core/arch/arm/dts/stm32mp211.dtsi
+++ b/core/arch/arm/dts/stm32mp211.dtsi
@@ -631,6 +631,11 @@
 			#address-cells = <1>;
 			#size-cells = <1>;
 
+			uid_otp@14 {
+				/* 0x14 -> otp5 */
+				reg = <0x14 0xc>;
+			};
+
 			part_number_otp@24 {
 				reg = <0x24 0x4>;
 			};
-- 
2.47.2

